emcc.py

```diff
@@ -1269,7 +1269,7 @@ def check(input_file):
         newargs += ['-pthread']
         # some pthreads code is in asm.js library functions, which are auto-exported; for the wasm backend, we must
         # manually export them
-        shared.Settings.EXPORTED_FUNCTIONS += ['_emscripten_get_global_libc', '___pthread_tsd_run_dtors', '__register_pthread_ptr', '_pthread_self']
+        shared.Settings.EXPORTED_FUNCTIONS += ['_emscripten_get_global_libc', '___pthread_tsd_run_dtors', '__register_pthread_ptr', '_pthread_self', '___emscripten_pthread_data_constructor']
```

Comment:
Its not clear to me exactly why this is needed.   `__emscripten_pthread_data_constructor` is defined in the same file a pthread_self so it should already be pulled in by in any program that calls pthread_self

Reply:
That file has `pthread_self` behind an ifdef - it's implemented there for non-pthreads builds it seems.

For pthreads builds, we define `pthread_self` in JS it appears https://github.com/emscripten-core/emscripten/blob/incoming/src/library_pthread.js#L844